### PR TITLE
Fix intersection with 0-area elements

### PR DESCRIPTION
**Pull request by seanjohnson08**
In a previous commit, we thought this issue was addressed: fba08c7

However, on trying to implement this in a downstream app - we discovered it wasn't fixed.

The test was passing because the test runner was overriding the 0x0 dimensions to 0x10, and did not sufficiently cover the edge case.

I've updated the calculateIsIntersecting method to accurately handle elements of all dimensions, and have removed the override from the test runner so that the test passes after this change and fails before.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "spaniel",
-  "version": "2.5.6",
+  "version": "2.5.7",
```
Comment (lynchbomb) [inline package.json]: why are we bumping the version within this PR? ideally we create a separate PR for the patch release.
